content/en-us/build/getting-started/run-a-local-development-network.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Run Your Own Local Testnet"
+weight: 2005
+menu:
+  build:
+    parent: "phat-basic"
+---
+
+
+## Overview
+
+In this tutorial, we're going to set up a development environment. We are going to deploy a full stack of the core blockchain and connect the Web UI to the blockchain. By the end of the tutorial, you will be able to:
+
+- Send confidential Commands and Queries
+- Get a ready-to-hack version of Phala Network for building your confidential DApps
+
+A full Phala Network stack has three components, with an optional Javascript SDK. The core components are available at [Phala-Network/phala-blockchain](https://github.com/Phala-Network/phala-blockchain):
+
+- `phala-node`: The Substrate blockchain node
+- `pRuntime`: The TEE runtime. Contracts run in `pRuntime`
+- `pherry`: The Substrate-TEE bridge relayer. Connects the blockchain and `pRuntime`
+
+<img src="/images/docs/developer/simple_architecture.png" alt="drawing" class="center"/>
+
+(Phala architecture overview)
+
+The Javascript SDK is at [Phala-Network/js-sdk](https://github.com/Phala-Network/js-sdk). The Web UI based on our SDK needs to connect to both the blockchain and the `pRuntime` to send Commands and Queries.
+
+## Setting up
+
+In this tutorial, we assume the operating system is *Ubuntu 22.04*. Other Linux distributions should also work, but the instructions or commands may vary.
+4 cores and 8GB RAM is the minimal requirement to build the project including the core blockchain.
+
+### Build from source
+
+The Phala-Network/phala-blockchain repository always contains [the latest build instructions](https://github.com/Phala-Network/phala-blockchain#native-build), at the time of writing (December 26, 2022), we use the following commands to set up development environment:
+
+
+```bash
+# First clone the repository
+git clone https://github.com/Phala-Network/phala-blockchain.git
+# Change to the repository directory
+cd phala-blockchain
+# Install system dependencies:
+sudo apt install -y build-essential pkg-config libssl-dev protobuf-compiler
+# Install Rust
+curl https://sh.rustup.rs -sSf | sh
+# Install dependencies for Substrate development
+git submodule update --init
+sh ./scripts/init.sh
+# Installl LLVM 14
+wget https://apt.llvm.org/llvm.sh
+chmod +x llvm.sh
+./llvm.sh 14
+```
+
+Then run the following command to build the Phala blockchain:
+
+```bash
+cargo build --release
+```
+
+It may take about 30 minutes to finish.
```

Comment:
Can we give a general hardware config as a reference here? like "30 minutes using Intel i5 XXXX with Y cores and Z threads"